samcli/lib/config/samconfig.py

```diff
@@ -106,19 +114,21 @@ def put(self, cmd_names, section, key, value, env=DEFAULT_ENV):
 
         if not self.document:
             self._read()
-            # Empty document prepare the initial structure.
-            # self.document is a nested dict, we need to check each layer and add new tables, otherwise duplicated key
-            # in parent layer will override the whole child layer
-            if self.document.get(env, None):
-                if self.document[env].get(self._to_key(cmd_names), None):
-                    if not self.document[env][self._to_key(cmd_names)].get(section, None):
-                        self.document[env][self._to_key(cmd_names)].update({section: {key: value}})
+        # Empty document prepare the initial structure.
+        # self.document is a nested dict, we need to check each layer and add new tables, otherwise duplicated key
+        # in parent layer will override the whole child layer
+        if self.document.get(env, None):
+            if self.document[env].get(self._to_key(cmd_names), None):
```

Comment:
Same here, I would put `self.document[env].get(self._to_key(cmd_names), None)` in a variable

Reply:
Rewrite this part to make the if-else branches cleaner.